glean-core/ios/GleanTests/HttpPingUploaderTests.swift

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import XCTest
+@testable import Glean
+
+class HttpPingUploaderTests: XCTestCase {
+    private let testPath = "/some/random/path/not/important"
+    private let testPing = "{ 'ping': 'test' }"
+    private let testConfig = Configuration(
+        userAgent: "Glean/Test 25.0.2",
+        connectionTimeout: 3050,
+        readTimeout: 7050,
+        pingTag: "Tag"
+    )
+    
+    func testUploadSuccess() {
+        class MockUploader: HttpPingUploader {
+            override func upload(path: String, data: String, config: Configuration, callback: @escaping (Bool, Error?) -> Void) {
+                super.upload(path: path, data: data, config: config, callback: callback)
```

Comment:
Will this try to send an actual HTTP request during testing?

Reply:
Not yet, I wasn't quite done with this one and needed to research how to set up a mock HTTP server like we do in Android.

Reply:
I've now updated this to use `OHHTTPStub` similar to `/mozilla-mobile/telemetry-ios` to better test the server responses.  This also prompted me to add a carthage Cartfile to import the dependency as well as some project settings changes to hook in the `carthage` command.  

I also had to add a `.swiftlint.yml` file in order to exclude the output director for the `OHHTTPStub` framework so their linting issues won't affect us.

Reply:
It looks like `telemetry-ios` uses a cool trick to only import the `OHHTTPStub` framework on tests, I'll have to ask Garvan about that.